packages/common/src/core/schemas/internal/applyUiSchemaElementOptions.ts

```diff
@@ -0,0 +1,35 @@
+import { deepFind } from "../../../objects/deepFind";
+import { cloneObject } from "../../../util";
+import { IUiSchema, UiSchemaElementOptions } from "../types";
+
+/**
+ * Apply a set of run-time configurations into the UI schema so
+ * the controls can be configured for the current environment.
+ * TODO: Add Example
+ * @param schema
+ * @param options
+ * @returns
+ */
+export function applyUiSchemaElementOptions(
```

Comment:
I found another edge case we need to account for here 😬 In the `uiSchema`, we can define "rules" to conditionally show/hide/disable fields based on the value of another field. A rule looks something like:

```ts
const uiSchema = {
   type: "Layout",
   elements: [
      ...,
      {
         type: "Control",
         labelKey: "some.label.key",
         scope: "/properties/description",
         rule: {
            effect: "DISABLE",
            condition: {
               scope: "/properties/name",
               schema: { enum: [ "hello" ] }
            }
         }
      }
   ]
}
```
This essentially says, if the value of the "name" field is "hello", disable the "description" field. 

This poses a problem though because we don't want this function to match `elementOptions.scope` with a rule's condition scope and spread the options in there.

Reply:
I've addressed this, but it's specific to this case. If we run into more cases, I'll write some more specific functions that are specific to the schema structures.